### DISCUSSION TITLE
Type hints lost when using immer middleware (v5.0.0)

### DISCUSSION THREAD

**Discussion by yuri2peter** [pmndrs/zustand#2819] [Bug report]
### Bug Description

I'm experiencing a type inference issue when using the immer middleware with Zustand v5.0.0. When using the middleware, the type of the draft object is inferred as any, leading to type errors.

This issue doesn't occur when using Zustand v4.x.

![image](https://github.com/user-attachments/assets/e90fef4b-4f95-4fcf-9b80-ae5528482705)

### Reproduction Link

https://stackblitz.com/edit/stackblitz-starters-xysert?file=app%2Ftest.ts&view=editor

**Comment by dai-shi** [comment]
Is it related with #2696? Can @chrisvander take a look please?

**Reply by chrisvander** [reply]
You're right, it's related. For some reason, when we do `SetStateType<A2>`, it resolves correctly to `{ count: number }` in all but the `Draft</* ... */>` type. In that, I'm seeing TypeScript inference calling it `Draft<Exclude<A2[0], (...args: any[]) => any>>` instead of resolving the inner type to `{ count: number }`.

I'll try to figure out why that might be.

**Reply by adriancalavie** [reply]
Any news on this?

**Reply by cflorez-nex** [reply]
Any news?

**Reply by CrazyClicker** [reply]
Still there

**Reply by dbritto-dev-seatpick** [reply]
would you mind creating a minimal repro on stackblitz?

**Reply by CrazyClicker** [reply]
Sure, https://stackblitz.com/edit/vitejs-vite-6zggesoa?file=src%2Fmain.ts
<img width="632" alt="image" src="https://github.com/user-attachments/assets/3fff843b-d492-43c0-b2fa-bffa7c18df97" />

**Reply by CrazyClicker** [reply]
My bad, problem was that immer was not explicitly added to project dependencies. It is all ok now
